src/groupByKey.ts

```diff
@@ -2,6 +2,24 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce(
+    (acum: {}, item: T) => {
+      const NameKey = String(item[key]);
+
+      if (NameKey in acum) {
+        return acum;
+      }
+
+      const newT = {
```

Comment:
what does newT mean? this name of variable doesn't tell me anything about what's inside of it